src/FondOfSpryker/Yves/GoogleTagManager/Plugin/VariableBuilder/TransactionProductVariables/TransactionProductFieldPriceExcludingTaxPlugin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManager\Plugin\VariableBuilder\TransactionProductVariables;
+
+use Exception;
+use FondOfSpryker\Yves\GoogleTagManager\Dependency\VariableBuilder\TransactionProductFieldPluginInterface;
+use Generated\Shared\Transfer\GooleTagManagerTransactionProductTransfer;
+use Generated\Shared\Transfer\ItemTransfer;
+use Spryker\Shared\Log\LoggerTrait;
+use Spryker\Yves\Kernel\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Yves\GoogleTagManager\GoogleTagManagerFactory getFactory()
+ */
+class TransactionProductFieldPriceExcludingTaxPlugin extends AbstractPlugin implements TransactionProductFieldPluginInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @param \Generated\Shared\Transfer\GooleTagManagerTransactionProductTransfer $gooleTagManagerTransactionProductTransfer
+     * @param \Generated\Shared\Transfer\ItemTransfer $itemTransfer
+     * @param array $params
+     *
+     * @return \Generated\Shared\Transfer\GooleTagManagerTransactionProductTransfer
+     */
+    public function handle(
+        GooleTagManagerTransactionProductTransfer $gooleTagManagerTransactionProductTransfer,
+        ItemTransfer $itemTransfer,
+        array $params = []
+    ): GooleTagManagerTransactionProductTransfer {
+        try {
+            $moneyPlugin = $this->getFactory()->getMoneyPlugin();
+
+            if ($itemTransfer->getUnitPrice()) {
+                return $gooleTagManagerTransactionProductTransfer->setPriceExcludingTax(
+                    $moneyPlugin->convertIntegerToDecimal($itemTransfer->getUnitPrice())
+                );
+            }
+
+            $priceExcludingTax = $itemTransfer->getUnitPrice() - $itemTransfer->getUnitTaxAmount();
```

Comment:
Wenn `getUnitPrice()` vorher false ist, kann das dann hier funktionieren?

Reply:
UnitPrice kann nicht false sein, allerdings "null". 

cast "null" to int(0)?

Reply:
Am besten fix einen test dafür schreiben, dann sollte es klarer sein. Darf das negativ sein zB.